src/scalar.rs

```diff
@@ -624,9 +624,12 @@ impl Scalar {
         res
     }
 
-    /// Computes the multiplicative inverse of this element,
-    /// failing if the element is zero.
+    /// Computes the multiplicative inverse of this element, failing if the
+    /// element is zero.
     pub fn invert(&self) -> CtOption<Self> {
+        if *self == Scalar::one() || *self == -Scalar::one() {
+            return CtOption::new(*self, !self.ct_eq(&Self::zero()));
```

Comment:
We need to keep in mind that this is no longer a constant time operation. My suggestion:

1) change the signature to `fn invert(&self) -> Option<Self>`
2) Create a new function `fn invert_ct(&self) -> CtOption<Self>` that will have the previous implementation